core-java-modules/core-java-regex/src/test/java/com/baeldung/regex/countmatches/CountMatchesUnitTest.java

```diff
@@ -0,0 +1,83 @@
+package com.baeldung.regex.countmatches;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.junit.Test;
+
+/**
+ * Unit Test intended to count number of matches of a RegEx using Java 8 and 9.
+ * 
+ * Java 9 is needed to run the test.
+ */
+public class CountMatchesUnitTest {
+
+    private static final Pattern EMAIL_ADDRESS_PATTERN = Pattern.compile("([a-z0-9_.-]+)@([a-z0-9_.-]+[a-z])");
+    private static final String TEXT_CONTAINING_EMAIL_ADDRESSES = "You can contact me through: writer@baeldung.com, editor@baeldung.com and team@bealdung.com";
+    private static final String TEXT_CONTAINING_FIVE_EMAIL_ADDRESSES = "Valid emails are: me@gmail.com, you@baeldung.com, contact@hotmail.com, press@anysite.com and support@bealdung.com";
+
+    @Test
+    public void givenContainingEmailString_whenJava8Match_thenCountMacthesFound() {
+        Matcher countEmailMatcher = EMAIL_ADDRESS_PATTERN.matcher(TEXT_CONTAINING_EMAIL_ADDRESSES);
+
+        int count = 0;
+        while (countEmailMatcher.find()) {
+            count++;
+        }
+
+        assertEquals(3, count);
+    }
+
+    @Test
+    public void givenContainingFiveEmailsString_whenJava8Match_thenCountMacthesFound() {
+        Matcher countFiveEmailsMatcher = EMAIL_ADDRESS_PATTERN.matcher(TEXT_CONTAINING_FIVE_EMAIL_ADDRESSES);
+
+        int count = 0;
+        while (countFiveEmailsMatcher.find()) {
+            count++;
+        }
+
+        assertEquals(5, count);
+    }
+
+    @Test
+    public void givenStringWithoutEmails_whenJava8Match_thenCountMacthesNotFound() {
+        Matcher noEmailMatcher = EMAIL_ADDRESS_PATTERN.matcher("Simple text without emails.");
+
+        int count = 0;
+        while (noEmailMatcher.find()) {
+            count++;
+        }
+
+        assertEquals(0, count);
+    }
+
+    @Test
+    public void givenContainingEmailString_whenStartingInJava9Match_thenCountMacthesFound() {
+        // Matcher countEmailMatcher = EMAIL_ADDRESS_PATTERN.matcher(TEXT_CONTAINING_EMAIL_ADDRESSES);
```

Comment:
Please put a line in line `// uncomment to try with Java 9` on all these commented out tests

Reply:
Done